python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -305,6 +331,54 @@ def get_actions(self, data):
             data,
         )
 
+        # The first thing we do is check to see if we're dealing with a
+        # classic SGTK setup. In that case, we're going to short-circuit
+        # the get_actions call and go into a legacy setup that makes use
+        # of the "tank" command by way of this api's process_manager.
+        config_entities = [d["entity"] for i, d in all_pc_data.iteritems()]
+        legacy_config_data = dict()
+
+        # A classic-style config will contain a roots.yml file that defines
+        # a set of storage roots. We can access that by way of the descriptor
+        # object that we already have. If there are required storages, then
```

Comment:
this roots.yml check isn't a good way to check if a config is classic or not. @jfboismenu @thebeeland maybe the `resolve_descriptor()` method on bootstrap should instead return something like a `Configuration` object which contains methods and properties to resolve descriptor, indicate if a config is classic etc? Inside bootstrap we have a `Configuration` object hierarchy to expose that - maybe we should consider exposing that? That would require some discussion...